RNA-seq/00c-tximeta_other_species.Rmd

```diff
@@ -0,0 +1,214 @@
+
+```{r sf_files}
+# Let's look at the full path for the quant.sf files
+sf_files
+```
+
+Let's extract the _sample_ names from the **file paths** using the `stringr` package.
+
+Notice how the file path is separated by `/`.
+If we were to split up this character string by `/`, the second to last item is the sample names (because we used them as directory names for the `salmon` output).
+This is exactly what `stringr::word()` allows us to do: split up the file paths by `/` and extract the sample names.
+
+```{r sample_names}
+sample_names <- stringr::word(sf_files, -2, sep = "/")
+sample_names
+```
+
+## Set up metadata
+
+`tximeta` needs a data frame with at least these two columns: 
+- a `files` column  with the file paths to the quant.sf files
+- a `names` column with the sample names
+
+```{r names_sf_files}
+coldata <- data.frame(files = sf_files, 
+                      names = sample_names)
+```
+
+We have more information about these samples stored in the metadata file that we will also want stored in `coldata`. 
+Let's read in the sample metadata from the TSV file and do a bit of set up. 
+We really only want two columns from this metadata file, so we will use a `dplyr::select()` to keep only those. 
+
+```{r sample_meta_df}
+# Read in the sample metadata TSV file and have a look
+sample_meta_df <- readr::read_tsv(meta_file) %>% 
+  # Select the two columns we need here
+  dplyr::select(refinebio_accession_code, refinebio_title)
+
+# Take a look at the preview
+sample_meta_df
+```
+
+We'll want this information to be added to the `coldata`, which we can do by using a join function to match up the rows between the two data frames and combine them. 
+
+```{r join-sample_meta_df}
+coldata <- coldata %>%
+  dplyr::inner_join(sample_meta_df, by = c("names" = "refinebio_accession_code"))
+
+coldata
+```
+
+## Set up reference data for a non-human, non-mouse species 
+
+```{r}
+indexDir <- file.path(dir, "Dm.BDGP6.22.98_salmon-0.14.1")
+```
+
+
+You can go to Ensembl to see the full list of species: http://ftp.ensembl.org/pub/release-98/fasta/
+Click through the file paths to find the file you want to use for annotation. 
+Right click on the file and copy its link address
+We will store the `base_url` where all the species FASTA files are located and then the second part of the url is species specific. 
+
+
+```{r}
+base_url <- "ftp://ftp.ensembl.org/pub/release-98/fasta/"
+
+fasta_ftp <- c(
+  paste0(base_url, "danio_rerio/cdna/Danio_rerio.GRCz11.cdna.all.fa.gz"),
+  paste0(base_url, "danio_rerio/ncrna/Danio_rerio.GRCz11.ncrna.fa.gz")
+)
+```
+
+Now you can follow a similar series of steps to obtain the Gene Transfer File (GTF) using this address: 
+http://ftp.ensembl.org/pub/release-103/gtf/
+
+```{r}
+base_url <- "ftp://ftp.ensembl.org/pub/release-103/gtf/"
+
+gtf_path <- paste0(base_url, "danio_rerio/Danio_rerio.GRCz11.103.gtf.gz")
+```
+
+Create a linkedTxome with the 
+
+```{r}
+makeLinkedTxome(indexDir = index_dir,
```

Comment:
I am unsure if these link-data.sh is working right. I'm getting this error when trying to use the zebrafish salmon index: 
```
Error: lexical error: invalid char in json text. index/Danio_rerio/header.json (right here) ------^
```